pkg/client/userd/commands/gather_traces.go

```diff
@@ -71,7 +71,16 @@ func (*traceCommand) tracesFor(ctx context.Context, conn *grpc.ClientConn, ch ch
 	ctx, span := otel.GetTracerProvider().Tracer("").Start(ctx, "tracesFor", trace.WithAttributes(attribute.String("component", component)))
 	defer span.End()
 	cli := common.NewTracingClient(conn)
-	result, err := cli.DumpTraces(ctx, &emptypb.Empty{})
+	cfg := client.GetConfig(ctx)
+	maxRecSize := int64(1024 * 1024 * 20) // Default to 20 Mb here. There might be a lot of traces.
+	if !cfg.Grpc.MaxReceiveSize.IsZero() {
```

Comment:
whoops, i realized there was a MaxReceive on grpc servers, but i had no idea it impacted clients, good catch